### PR TITLE
Bump elastic stack version automation and auto-merge with mergify

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

1) Automerge with `mergify` if CI Passes for all those PRs with a new dependency.
2) Script to bump the versions:

```
$ .ci/bump-stack-version.sh 8.0.0-3beeb19e
Update stack with version 8.0.0-3beeb19e in docker-compose.yml
Update stack with version 8.0.0-3beeb19e in deployment.yaml
```

returns:

```diff
diff --git a/cli/config/compose/profiles/fleet/docker-compose.yml b/cli/config/compose/profiles/fleet/docker-compose.yml
index 0af295c..981abf7 100644
--- a/cli/config/compose/profiles/fleet/docker-compose.yml
+++ b/cli/config/compose/profiles/fleet/docker-compose.yml
@@ -16,7 +16,7 @@ services:
       - xpack.security.authc.api_key.enabled=true
       - ELASTIC_USERNAME=elastic
       - ELASTIC_PASSWORD=changeme
-    image: "docker.elastic.co/elasticsearch/elasticsearch:${stackVersion:-8.0.0-3beeb19e-SNAPSHOT}"
+    image: "docker.elastic.co/elasticsearch/elasticsearch:${stackVersion:-8.0.0-SNAPSHOT}"
     platform: ${stackPlatform:-linux/amd64}
     ports:
       - "9200:9200"
@@ -28,7 +28,7 @@ services:
       test: "curl -f http://localhost:5601/login | grep kbn-injected-metadata 2>&1 >/dev/null"
       retries: 600
       interval: 1s
-    image: "docker.elastic.co/${kibanaDockerNamespace:-kibana}/kibana:${kibanaVersion:-8.0.0-3beeb19e-SNAPSHOT}"
+    image: "docker.elastic.co/${kibanaDockerNamespace:-kibana}/kibana:${kibanaVersion:-8.0.0-SNAPSHOT}"
     platform: ${stackPlatform:-linux/amd64}
     ports:
       - "5601:5601"
@@ -48,7 +48,7 @@ services:
       - "XPACK_FLEET_AGENTS_FLEET_SERVER_HOSTS=[\"http://fleet-server:8220\"]"
       - "XPACK_FLEET_AGENTS_TLSCHECKDISABLED=true"
   fleet-server:
-    image: "docker.elastic.co/beats/elastic-agent:${stackVersion:-8.0.0-3beeb19e-SNAPSHOT}"
+    image: "docker.elastic.co/beats/elastic-agent:${stackVersion:-8.0.0-SNAPSHOT}"
     depends_on:
       elasticsearch:
         condition: service_healthy
diff --git a/cli/config/compose/profiles/metricbeat/docker-compose.yml b/cli/config/compose/profiles/metricbeat/docker-compose.yml
index e87a416..a4ff128 100644
--- a/cli/config/compose/profiles/metricbeat/docker-compose.yml
+++ b/cli/config/compose/profiles/metricbeat/docker-compose.yml
@@ -9,7 +9,7 @@ services:
       - xpack.monitoring.collection.enabled=true
       - ELASTIC_USERNAME=elastic
       - ELASTIC_PASSWORD=changeme
-    image: "docker.elastic.co/elasticsearch/elasticsearch:${stackVersion:-8.0.0-3beeb19e-SNAPSHOT}"
+    image: "docker.elastic.co/elasticsearch/elasticsearch:${stackVersion:-8.0.0-SNAPSHOT}"
     platform: ${stackPlatform:-linux/amd64}
     ports:
       - "9200:9200"
diff --git a/cli/config/kubernetes/base/elasticsearch/deployment.yaml b/cli/config/kubernetes/base/elasticsearch/deployment.yaml
index 6b47cf2..271cb07 100644
--- a/cli/config/kubernetes/base/elasticsearch/deployment.yaml
+++ b/cli/config/kubernetes/base/elasticsearch/deployment.yaml
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
       - name: elasticsearch
-        image: docker.elastic.co/elasticsearch/elasticsearch:8.0.0-3beeb19e-SNAPSHOT
+        image: docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT
         envFrom:
           - configMapRef:
               name: elasticsearch-config
diff --git a/cli/config/kubernetes/base/fleet-server/deployment.yaml b/cli/config/kubernetes/base/fleet-server/deployment.yaml
index 45ff9d9..94597cc 100644
--- a/cli/config/kubernetes/base/fleet-server/deployment.yaml
+++ b/cli/config/kubernetes/base/fleet-server/deployment.yaml
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
       - name: fleet-server
-        image: docker.elastic.co/beats/elastic-agent:8.0.0-3beeb19e-SNAPSHOT
+        image: docker.elastic.co/beats/elastic-agent:8.0.0-SNAPSHOT
         env:
         - name: FLEET_SERVER_ENABLE
           value: "1"
diff --git a/cli/config/kubernetes/base/kibana/deployment.yaml b/cli/config/kubernetes/base/kibana/deployment.yaml
index d0daffc..abbdc1c 100644
--- a/cli/config/kubernetes/base/kibana/deployment.yaml
+++ b/cli/config/kubernetes/base/kibana/deployment.yaml
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
       - name: kibana
-        image: docker.elastic.co/kibana/kibana:8.0.0-3beeb19e-SNAPSHOT
+        image: docker.elastic.co/kibana/kibana:8.0.0-SNAPSHOT
         env:
         - name: ELASTICSEARCH_URL
           value: http://elasticsearch:9200

```

## Why is it important?

Handle the stack versions on a PR basis to ensure breaking changes are handled correctly. Unfortunately we cannot use `dependabot` since those versions are handled differently.

## Issues

Similar to https://github.com/elastic/beats/pull/25148

## Tasks

- [ ] Requires to normalise the branch strategy with the same as `beats` (7.x, 7.11, 7.12) instead of using `7.11.x`
- [x] Requires to find the sed to replace the docker-compose variable interpolation

### CODE REVIEW

File: .ci/bump-stack-version.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+"
+
+echo "Update stack with version ${VERSION}"
+for FILE in ${FILES} ; do
+	${SED} -E -e "s#(image: docker\.elastic\.co/.*):[0-9]+\.[0-9]+\.[0-9]+(-[a-f0-9]{8})?#\1:${VERSION}#g" $FILE
+	## TODO: sed "docker.elastic.co/kibana/kibana:${kibanaTag:-8.0.0-SNAPSHOT}" ??
+done
+
+echo "Commit changes"
+git config user.email
```
Comment (v1v) [inline .ci/bump-stack-version.sh:33]:
```suggestion
```

**Review by cachedout — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Review by mdelapenya — APPROVED** [review]
I like this! Let's merge it and use it as foundation to modify more files, like the Jenkinsfile.

**Comment by mdelapenya** [comment]
One thing to consider here: we have code to always update (docker pull) the images for the current branch (see https://github.com/elastic/e2e-testing/blob/master/e2e/_suites/fleet/ingest_manager_test.go#L132-L144) I guess that if we add this new feature, where the images will be tied to a commit, then that part won't be needed IIUC

**Comment by v1v** [comment]
> I guess that if we add this new feature, where the images will be tied to a commit, then that part won't be needed IIUC

This automation will only affect the below docker images:

https://github.com/elastic/e2e-testing/blob/88d8178c404789cbcf14c5f9b7a39baf8788a801/e2e/_suites/fleet/ingest_manager_test.go#L133-L136


Let's syncup offline and see if there is something else missing

File: .ci/bump-stack-version.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	SED="sed -i .bck"
+else
+	SED="sed -i"
+fi
+
+echo "Update stack with version ${VERSION} in docker-compose.yml"
+find . -name 'docker-compose.yml' -print0 |
+	while IFS= read -r -d '' FILE ; do
+		${SED} -E -e "s#(image: (\")?docker\.elastic\.co/.*):-[0-9]+\.[0-9]+\.[0-9]+(-[a-f0-9]{8})?#\1:-${VERSION}#g" $FILE
+	done
```
Comment (v1v) [inline .ci/bump-stack-version.sh:31]: Let's only search for the `profiles`, services are not needed at the moment

**Pull request merged**